tutorial/marine/quick_start.md

```diff
@@ -0,0 +1,296 @@
+✨   Done! New project created /workspaces/devcontainer/greeter
+root@56892f4726bb:/workspaces/devcontainer#
+```
+
+We initiated a module project call `greeter` which is now visible in the VSCode Explorer with the following directories and files:
+
+* scripts
+* src
+* Cargo.toml
+* Config.toml
+
+Let's unpack.
+
+The Cargo.tom file, see below, is pretty standard Rust fare except for:
+
+1. We need to specify the binary information for our wasm32-wasi compile target
+2. We add the [Fluence](https://github.com/fluencelabs/fluence) crate to the dependencies, which includes the marine-rs sdk and the Marine-specific Wasm macros
+3. We add the fluence-test crate to the dev dependencies which allows us to write "regular" Rust unit-tests that can be run with `cargo test`
+
+```toml
+# Cargo.toml
+[package]
+name = "greeter"
+version = "0.1.0"
+authors = ["boneyard93501 <4523011+boneyard93501@users.noreply.github.com>"]
+edition = "2018"
+description = "greeter, a Marine wasi module"
+license = "Apache-2.0"
+
+[[bin]]    <-- 1
+name = "greeter"
+path = "src/main.rs"
+
+[dependencies] <-- 2
+fluence = { version="0.6.9", features=["logger"]}
+log = "0.4.14"
+
+[dev-dependencies]  <-- 3
+fluence-test = "0.1.9"
+
+[dev]
+[profile.release]
+opt-level = "s"
+```
+
+The Config.toml file provides the meta data necessary to properly instrument a Wasm IT module for execution by the Marine runtime.
+
+```toml
+# Config.toml
+modules_dir = "artifacts/"   <-- 1
+
+[[module]]  <-- 2
+    name = "greeter"
+    mem_pages_count = 1
+    logger_enabled = true
+```
+
+As illustrated in Figure 1 above, the Config.toml file is required for the local execution of Wasm modules such as the Repl or unittests. A Config.toml files consists of:
+
+1. Location of the wasm module to be executed by Marine
+2. One or mode module specific runtime requirements such as module name, logging particulars, local access and memory allocation, Recall, [Wasm memory](https://webassembly.github.io/spec/core/exec/runtime.html#memory-instances) is allocated in linear pages of 64Ki each.
+
+As evident by the existence of the main.rs file, Marine Wasm modules are created as binaries and as we see below, the Marine toolchain compiles our Rust code to the desired wasm32-wasi target. Due to our the Wasm IT target, our Rust files look a little different than Rust application or even wasm-bindgen code.
+
+```rust
+// main.rs
+use fluence::{marine, module_manifest, WasmLoggerBuilder};  <-- 1
+
+module_manifest!();  <-- 2
+
+pub fn main() {
+    WasmLoggerBuilder::new().build().unwrap();  <-- 3
+}
+
+#[marine]  <-- 4
+pub fn greeting(name: String) -> String {  <-- 5
+    format!("Hi, {}", name)
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::greeting;
+    use fluence_test::marine_test; <-- 6
+
+    #[marine_test(config_path = "../Config.toml", modules_dir = "../artifacts")]
+    fn test_greeting() {
+        let name = "Marine".to_string();
+        let res = greeting(name.clone());
```

Comment:
The code doesn't compile:
```
warning: unused import: `crate::greeting`
  --> src/main.rs:32:9
   |
32 |     use crate::greeting;
   |         ^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

error[E0618]: expected function, found `__m_generated_greeting::MGeneratedStructgreeting`
  --> src/main.rs:38:19
   |
35 |     #[marine_test(config_path = "../Config.toml", modules_dir = "../artifacts")]
   |     ---------------------------------------------------------------------------- `greeting` has type `__m_generated_greeting::MGeneratedStructgreeting`
...
38 |         let res = greeting(name.clone());
   |                   ^^^^^^^^--------------
   |                   |
   |                   call expression requires function
```

That's because `#[marine_test]` uses not the code from `super::`, but instead it generates code from `.wasm` modules loaded from the FS. Note warning on unused `use crate::greeting`.

So it should be `greeting.greeting(name.clone())` where `greeting` is the wasm module name from _Config.toml_ and `.greeting()` is the function defined by that wasm module.

It works like this:
```suggestion
    use fluence_test::marine_test; <-- 6

    #[marine_test(config_path = "../Config.toml", modules_dir = "../artifacts")]
    fn test_greeting() {
        let name = "Marine".to_string();
        let res = greeting.greeting(name.clone());
```

Reply:
don't edit code in the browser without checking. 
that screw up the template notion since given the project an other name than "greeting" leads to a broken test. hm.